internal/cli/opsmanager/admin/backup/s3/create.go

```diff
@@ -0,0 +1,137 @@
+		S3BucketName:     opts.s3BucketName,
+		S3MaxConnections: opts.s3MaxConnections,
+		AcceptedTos:      &opts.acceptedTos,
+	}
+
+	if opts.disableProxyS3 {
+		blockstore.DisableProxyS3 = &opts.disableProxyS3
+	}
+
+	if opts.sseEnabled {
+		blockstore.SSEEnabled = &opts.sseEnabled
+	}
+
+	if opts.pathStyleAccessEnabled {
+		blockstore.PathStyleAccessEnabled = &opts.pathStyleAccessEnabled
+	}
+
+	return blockstore
+}
+
+// mongocli ops-manager admin backup blockstore(s) create [--assignment][--encryptedCredentials][--id id]
+// [--label label][--loadFactor loadFactor][--uri uri][--ssl][--writeConcern writeConcern] [--awsAccessKey awsAccessKey]
+// [--awsSecretKey awsSecretKey] [--s3AuthMethod s3AuthMethod] [--s3BucketEndpoint s3BucketEndpoint] [--s3BucketName s3BucketName]
+// [--s3MaxConnections s3MaxConnections] [--disableProxyS3 disableProxyS3] [--acceptedTos acceptedTos] [--sseEnabled sseEnabled]
+// [--pathStyleAccessEnabled pathStyleAccessEnabled]
+
+func CreateBuilder() *cobra.Command {
+	opts := &CreateOpts{}
+	opts.Template = createTemplate
+	cmd := &cobra.Command{
+		Use:   "create",
+		Short: create,
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			opts.OutWriter = cmd.OutOrStdout()
+			return opts.init()
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return opts.Run()
+		},
+	}
+
+	cmd.Flags().StringVar(&opts.ID, flag.ID, "", usage.BlockstoreID)
+	cmd.Flags().StringSliceVar(&opts.Label, flag.Label, []string{}, usage.Label)
+	cmd.Flags().BoolVar(&opts.Assignment, flag.Assignment, false, usage.Assignment)
+	cmd.Flags().BoolVar(&opts.EncryptedCredentials, flag.EncryptedCredentials, false, usage.EncryptedCredentials)
+	cmd.Flags().Int64Var(&opts.LoadFactor, flag.LoadFactor, 0, usage.LoadFactor)
+	cmd.Flags().StringVar(&opts.URI, flag.URI, "", usage.BlockstoreURI)
+	cmd.Flags().StringVar(&opts.WriteConcern, flag.WriteConcern, "", usage.WriteConcern)
+	cmd.Flags().StringVar(&opts.awsAccessKey, flag.AWSAccessKey, "", usage.AWSAccessKey)
+	cmd.Flags().StringVar(&opts.awsSecretKey, flag.AWSSecretKey, "", usage.AWSSecretKey)
+	cmd.Flags().StringVar(&opts.s3AuthMethod, flag.S3AuthMethod, "", usage.S3AuthMethod)
+	cmd.Flags().StringVar(&opts.s3BucketEndpoint, flag.S3BucketEndpoint, "", usage.S3BucketEndpoint)
+	cmd.Flags().StringVar(&opts.s3BucketName, flag.S3BucketName, "", usage.S3BucketName)
+	cmd.Flags().Int64Var(&opts.s3MaxConnections, flag.S3MaxConnections, 0, usage.S3MaxConnections)
+	cmd.Flags().BoolVar(&opts.disableProxyS3, flag.DisableProxyS3, false, usage.DisableProxyS3)
+	cmd.Flags().BoolVar(&opts.acceptedTos, flag.AcceptedTos, false, usage.AcceptedTos)
+	cmd.Flags().BoolVar(&opts.sseEnabled, flag.SSEEnabled, false, usage.SSEEnabled)
+	cmd.Flags().BoolVar(&opts.pathStyleAccessEnabled, flag.PathStyleAccessEnabled, false, usage.PathStyleAccessEnabled)
+
+	cmd.Flags().StringVarP(&opts.Output, flag.Output, flag.OutputShort, "", usage.FormatOut)
+
+	_ = cmd.MarkFlagRequired(flag.ID)
```

Comment:
is this mandatory for create? when you're creating you don't know the ID, or is this a name you're giving?

Reply:
yes it is, https://docs.opsmanager.mongodb.com/current/reference/api/admin/backup/snapshot/s3Configs/create-one-s3-blockstore-configuration/#request-body-parameters

<img width="731" alt="Screenshot 2020-10-12 at 18 01 07" src="https://user-images.githubusercontent.com/5663078/95772611-eaed5100-0cb4-11eb-975b-7f2202b52c87.png">

Reply:
do you know if the UI uses name vs id? I'd have a feeling the UI uses name and as such I'd rather have the flag be name and not id

Reply:
It uses name. I am going to change id to name
<img width="561" alt="Screenshot 2020-10-12 at 18 33 19" src="https://user-images.githubusercontent.com/5663078/95775001-69e48880-0cb9-11eb-8e11-8ccf352a0278.png">

Reply:
I am going to rename it for blockstore as well

Reply:
thanks

Reply:
you need to update the required to Name instead of ID